src/main/kotlin/com/fasterxml/jackson/module/kotlin/KotlinModule.kt

```diff
@@ -104,38 +124,29 @@ class KotlinModule constructor (
         var reflectionCacheSize: Int = 512
             private set
 
-        var nullToEmptyCollection: Boolean = false
-            private set
-
-        var nullToEmptyMap: Boolean = false
-            private set
-
-        var nullIsSameAsDefault: Boolean = false
-            private set
-
-        var singletonSupport = DISABLED
-            private set
-
-        var strictNullChecks = false
-            private set
-
-        fun reflectionCacheSize(reflectionCacheSize: Int) = apply { this.reflectionCacheSize = reflectionCacheSize }
+        private val bitSet: BitSet = 0.toBitSet().apply {
+            KotlinFeature.values().filter { it.enabledByDefault() }.forEach { or(it.bitSet) }
+        }
 
-        fun nullToEmptyCollection(nullToEmptyCollection: Boolean) =
-            apply { this.nullToEmptyCollection = nullToEmptyCollection }
+        fun withReflectionCacheSize(reflectionCacheSize: Int) = apply {
+            this.reflectionCacheSize = reflectionCacheSize
+        }
 
-        fun nullToEmptyMap(nullToEmptyMap: Boolean) = apply { this.nullToEmptyMap = nullToEmptyMap }
+        fun enable(feature: KotlinFeature) = apply {
+            bitSet.or(feature.bitSet)
+        }
 
-        fun nullIsSameAsDefault(nullIsSameAsDefault: Boolean) = apply { this.nullIsSameAsDefault = nullIsSameAsDefault }
+        fun disable(feature: KotlinFeature) = apply {
+            bitSet.andNot(feature.bitSet)
+        }
 
-        fun singletonSupport(singletonSupport: SingletonSupport) =
-            apply { this.singletonSupport = singletonSupport }
+        fun set(feature: KotlinFeature, enabled: Boolean) = when {
```

Comment:
No strong opinion here, but core Jackson uses "enable / disable / configure" names (that is, "configure()" instead of "set()"). I am fine with either one, just mentioning this.